### DISCUSSION TITLE
2.4" TFT LCD // ILI9325

### DISCUSSION THREAD

**Discussion by shri360** [Bodmer/TFT_eSPI#2496] [Compatible displays and setup files]
Hi there!

 I'm trying to interface 2.4" TFT LCD with ESP32-WROOM-32.

Lcd model,

![photo_6086669279723369451_y](https://user-images.githubusercontent.com/34938549/224726744-1b1087ce-32b6-474b-9099-c252363933a5.jpg)

Connections
TFT		ESP32	
LDC_RST	--	GPIO32	
LDC_CS	--	GPIO33	
LDC_RS	--	GPIO15
LDC_WR	--	GPIO4	
LDC_RD	--	GPIO2	
LDC_D0	--	GPIO12	
LDC_D1	--	GPIO13	
LDC_D2	--	GPIO26	
LDC_D3	--	GPIO25	
LDC_D4	--	GPIO17	
LDC_D5	--	GPIO16	
LDC_D6	--	GPIO27	
LDC_D7	--	GPIO14	
3.3V	--	3.3V	
GND	--	GND

After uploading code showing white screen. really don't know what I'm doing wrong.

**Comment by Bodmer** [comment]
The ILI9325 is not one of the supported displays. So it may not work.

Which driver have you used in the setup file?

Run the Read_User_Setup sketch and look at the serial output to see if the pins are correct.

**Reply by shri360** [reply]
@Bodmer thank you and I appreciate your response. 

Finally it's working!!
i used MCUFriend library and followed this guide. [https://forum.arduino.cc/t/help-mcufriend-3-5-tft-lcd-480x320-esp32-solved/559854/5?u=shripad_hegde](url)

**Reply by Gitman13264** [reply]
@shri360 @Bodmer  the link no longer works, can you provide me with any other information on how you got the ili9325 LCD with the ESP-32-WROOM-32. really struggling to find any library that works

**Reply by elgerg** [reply]
> @shri360 @Bodmer the link no longer works, can you provide me with any other information on how you got the ili9325 LCD with the ESP-32-WROOM-32. really struggling to find any library that works

Copy and paste the link. It works for me.

**Reply by shri360** [reply]
@Gitman13264 , for ili9325 Display MCUFriend Library worked for me..

Here is basic code for the ili9325 using MCUFriend Library..

`#define LCD_CS 33 // Chip Select goes to Analog 3
#define LCD_RS 15 // LCD_RS = Register Select or LCD_CD = Command/Data goes to Analog 2
#define LCD_WR 4 // LCD Write goes to Analog 1
#define LCD_RD 2 // LCD Read goes to Analog 0
#define LCD_RESET 32 // Can alternately just connect to Arduino's reset pinCan alternately just connect to Arduino's reset pin

#include <SPI.h>          // f.k. for Arduino-1.5.2
#include "Adafruit_GFX.h"// Hardware-specific library
#include <MCUFRIEND_kbv.h>
MCUFRIEND_kbv tft;
//#include <Adafruit_TFTLCD.h>
//Adafruit_TFTLCD tft(LCD_CS, LCD_CD, LCD_WR, LCD_RD, LCD_RESET);

// Assign human-readable names to some common 16-bit color values:
#define  BLACK   0x0000
#define BLUE    0x001F
#define RED     0xF800
#define GREEN   0x07E0
#define CYAN    0x07FF
#define MAGENTA 0xF81F
#define YELLOW  0xFFE0
#define WHITE   0xFFFF

void setup()
{
    Serial.begin(9600);
    uint16_t ID = tft.readID();
    if (ID == 0xD3) ID = 0x9481;
    tft.begin(ID);
    tft.setRotation(0);
}

void loop()
{
    tft.fillScreen(YELLOW);
    delay (1000);
    tft.fillScreen(BLACK);
    delay (500);
    tft.fillScreen(MAGENTA);
    delay (1500);
} `

**Comment by shri360** [comment]
@Gitman13264 
Here is the link.
post #5

https://forum.arduino.cc/t/help-mcufriend-3-5-tft-lcd-480x320-esp32-solved/559854/5


Thanks for this man on arduino forum 
https://forum.arduino.cc/u/wieb/

**Reply by Gitman13264** [reply]
okay link works now thank you so much, sorry to bother but did you change anything in the library files of MCu friend?

**Reply by shri360** [reply]
@Gitman13264   No need to change anything in the library..  make sure that same connections as mentioned in the example code that’s it.

#define LCD_CS 33 // Chip Select goes to Analog 3
#define LCD_RS 15 // LCD_RS = Register Select or LCD_CD = Command/Data goes to Analog 2
#define LCD_WR 4 // LCD Write goes to Analog 1
#define LCD_RD 2 // LCD Read goes to Analog 0
#define LCD_RESET 32 // Can alternately just connect to Arduino's reset pin

**Reply by Gitman13264** [reply]
no idea why this error keeps popping up, have uninstalled the library and installed again, thanks for all your help. very much appreciated, also was your chip specifically the ESP-32_Wroom32
![image](https://github.com/user-attachments/assets/4cf9599f-0c03-4287-9dba-2354d1446d51)

**Reply by RishitShetty** [reply]
Hi, Did you find any solution for that problem, even I am experiencing similliar issues

**Comment by Gitman13264** [comment]
Yes, i figured it out eventually. Glad to help give me more information
